samza-api/src/main/java/org/apache/samza/table/descriptors/RemoteTableDescriptor.java

```diff
@@ -243,29 +239,52 @@ public String getProviderFactoryClassName() {
         throw new SamzaException("Failed to create default rate limiter", ex);
       }
       addTableConfig(RATE_LIMITER, SerdeUtils.serialize("rate limiter", defaultRateLimiter), tableConfig);
+      if (defaultRateLimiter instanceof TablePart) {
+        tableConfig.putAll(((TablePart) defaultRateLimiter).toConfig(jobConfig, new MapConfig(tableConfig)));
+      }
     } else if (rateLimiter != null) {
       addTableConfig(RATE_LIMITER, SerdeUtils.serialize("rate limiter", rateLimiter), tableConfig);
+      if (rateLimiter instanceof TablePart) {
+        tableConfig.putAll(((TablePart) rateLimiter).toConfig(jobConfig, new MapConfig(tableConfig)));
+      }
     }
 
-    // Serialize the readCredit functions
+    // Handle readCredit functions
     if (readCreditFn != null) {
       addTableConfig(READ_CREDIT_FN, SerdeUtils.serialize("read credit function", readCreditFn), tableConfig);
+      tableConfig.putAll(readCreditFn.toConfig(jobConfig, new MapConfig(tableConfig)));
     }
-    // Serialize the writeCredit functions
+
+    // Handle writeCredit functions
     if (writeCreditFn != null) {
       addTableConfig(WRITE_CREDIT_FN, SerdeUtils.serialize("write credit function", writeCreditFn), tableConfig);
+      tableConfig.putAll(writeCreditFn.toConfig(jobConfig, new MapConfig(tableConfig)));
     }
 
+    // Handle read retry policy
     if (readRetryPolicy != null) {
       addTableConfig(READ_RETRY_POLICY, SerdeUtils.serialize("read retry policy", readRetryPolicy), tableConfig);
+      tableConfig.putAll(readRetryPolicy.toConfig(jobConfig, new MapConfig(tableConfig)));
     }
 
+    // Handle write retry policy
     if (writeRetryPolicy != null) {
       addTableConfig(WRITE_RETRY_POLICY, SerdeUtils.serialize("write retry policy", writeRetryPolicy), tableConfig);
+      tableConfig.putAll(writeRetryPolicy.toConfig(jobConfig, new MapConfig(tableConfig)));
     }
 
     addTableConfig(ASYNC_CALLBACK_POOL_SIZE, String.valueOf(asyncCallbackPoolSize), tableConfig);
 
+    // Handle table reader function
+    addTableConfig(READ_FN, SerdeUtils.serialize("read function", readFn), tableConfig);
```

Comment:
Do we need a null check for readFn?

Reply:
No, there is a check to ensure it's not null.